### DISCUSSION TITLE
Can .any (https://vitaly-t.github.io/pg-promise/Database.html#any) return an array with falsy value ? eg : [undefined] [null] if so, in which case ?

### DISCUSSION THREAD

**Discussion by ChaitanyaNirf** [vitaly-t/pg-promise#942] [Q&A]
// API: http://vitaly-t.github.io/pg-promise/Database.html#any
        any<T = any>(query: QueryParam, values?: any): XPromise<T[]>

**Comment by vitaly-t** [comment]
It returns an array with values, as per your multi-row query result. What is the issue?

**Reply by ChaitanyaNirf** [reply]
facing an intermittent issue with a db call that returns an array. I have a condition to check if this data is an array and if it's length is greater than 0. If either condition isn't met, I am simply resolving an empty array. However, sometimes this check is being bypassed, causing errors in further executions...
Based on the error in the further execution, it seems that array contains a falsy value. 
Something like this => [undefined] 
It bypasses the if condition as it is an array and it's length is also greater than 0

Just want to understand, if it is possible for  .any() to return [undefined] 
// API: http://vitaly-t.github.io/pg-promise/Database.html#any
    any<T = any>(query: QueryParam, values?: any): XPromise<T[]>

**Comment by vitaly-t** [comment]
For starters, you need to see what query you are executing when undefined is returned. Otherwise, it is impossible to know where the issue is. Once you have the query, try to execute it externally, see what you get. Theoretically, it is possible for a query to return an undefined.

**Reply by ChaitanyaNirf** [reply]
Thanks, will definitely try that the next time it happens.

**Reply by vitaly-t** [reply]
Still nothing? ;) If the issue is resolved, then please close this discussion ;)

**Comment by vitaly-t** [comment]
Closing, due to no response or updates from the author.
